UsingLinkClicky.md

```diff
@@ -0,0 +1,396 @@
+# Using Behat
+
+The purpose of testing is to ensure that requirements are met. Additionally regression testing ensures that changes to the code or the site do not make unexpected changes elsewhere. Best practices dictate that we want full testing coverage, yet we want our set of tests DRY (don’t repeat yourself). So, what are the best practices for testing?
+
+Also see [Writing BDD tests for Drupal.org](https://www.drupal.org/node/1578324), [Feature/Scenario Review Checklist](https://www.drupal.org/node/1782858), 
+
+## Analysis
+
+Let’s examine the various considerations of testing our products and sites.
+
+### Functionality
+
+ In our product development we need to test functionality of the site. For our websites this includes:
+
+1. Behaviour: Does the site behave or function as expected?
+
+2. Theme: Does the site have the correct design, layouts, fonts, and colors? This includes menus, block placement, logos, etc.
+
+3. Content: Did our placeholder content and help text get imported properly?
+
+4. Hosting Environment: Where is this site hosted? Is it local on your machine? Is it on sites? On Anchorage? 
+
+5. Authentication Scheme: What approach do use to authenticate? Native Drupal? Webauth? SamL?
+
+### Lifecycle
+
+Testing for our product development and subsequent sites has a sort of timeline or lifecycle. Which looks roughly like this:
+
+    Requirements
+        > Development
+            > Deployment
+                > Customization
+                    > Production
+                        > Launch
+                            > Live
+                                
+                           
+For each phase of this lifecycle, we may need a specific set of tests. Although individual tests may overlap between phases, there are many tests that are only relevant to a particular phase. 
+
+#### Requirements
+
+In the requirements phase we define the functionality. This functionality might be for a new or updated product, theme, behavior, content, etc. Ideally, this is the phase in which we identify and draft our first Behat tests. 
+
+#### Development
+
+During development we are adding to the functionality of the product, so here’s where we thoroughly test functionality. Since you can expect that the site will get rebuilt, anything goes. In our tests, we can add random content, delete existing pages, change the name of the site, etc.
+
+#### Deployment
+
+The deployment phase signals that we are no longer in development. When we deploy a site for a customer, we need to assume that we fully tested product functionality in development. In this phase we want to ensure that the site was deployed properly. We need to verify that the modules, themes, and content were all enabled and imported properly. We want happy customers!
```

Comment:
This is not always true. We don't always deploy a fully functional website. Most of the time we deploy a website and then iterate on.